lib/ember-test-helpers/test-module.js

```diff
@@ -107,7 +107,8 @@ export default Klass.extend({
     function nextStep() {
       var step = steps.shift();
       if (step) {
-        return Ember.RSVP.resolve(step.call(context)).then(nextStep);
+        // guard against exceptions, for example missing components referenced from needs.
+        return new Ember.RSVP.Promise(function(ok) {ok(step.call(context));}).then(nextStep);
```

Comment:
Yes, this is roughly what I was thinking also (though with .catch instead of two args to .then).